imap_processing/tests/idex/test_idex_l1a.py

```diff
@@ -104,6 +103,49 @@ def test_idex_tof_high_data_from_cdf(decom_test_data: xr.Dataset):
     assert (l1_data["TOF_High"][13].data == data).all()
 
 
+def test_validate_l1a_idex_data_variables(
+    decom_test_data: xr.Dataset, l1a_example_data: xr.Dataset
+):
+    """
+    Verify that each of the 6 waveform and telemetry arrays are equal to the
+    corresponding array produced by the IDEX team using the same l0 file.
+
+
+    Parameters
+    ----------
+    decom_test_data : xarray.Dataset
+        The dataset to test with
+    l1a_example_data: xarray.Dataset
+        A dataset containing the 6 waveform and telemetry arrays
+    """
+    # Lookup table to match the SDS array names to the Idex Team array names
+    match_variables = {
+        "TOF L": "TOF_Low",
+        "TOF H": "TOF_High",
+        "TOF M": "TOF_Mid",
+        "Target H": "Target_High",
+        "Target L": "Target_Low",
+        "Ion Grid": "Ion_Grid",
+        "Time (high sampling)": "time_high_sample_rate",
+        "Time (low sampling)": "time_low_sample_rate",
+    }
+    # The Engineering data is converting to UTC, and the SDC is converting to J2000,
+    # for 'epoch' and 'Timestamp' so this test is using the raw time value 'SCHOARSE' to
+    # validate time
+    arrays_to_skip = ["Timestamp", "Epoch", "event"]
+
+    # loop through all keys from the l1a example dict
+    for var in l1a_example_data.variables:
+        if var not in arrays_to_skip:
+            # Find the corresponding array name
+            cdf_var = match_variables.get(var, var.lower())
+
+            assert np.array_equal(decom_test_data[cdf_var], l1a_example_data[var]), (
```

Comment:
Numpy has nice pretty prints on the mismatched values, so we should use their testing assertion functions when possible.
```suggestion
            np.testing.assert_array_equal(decom_test_data[cdf_var], l1a_example_data[var])
```